aiida/engine/runners.py

```diff
@@ -263,27 +261,35 @@ def run_get_pk(self, process, *args, **inputs):
         result, node = self._run(process, *args, **inputs)
         return ResultAndPk(result, node.pk)
 
-    def call_on_calculation_finish(self, pk, callback):
+    def call_on_process_finish(self, pk, callback):
         """
-        Callback to be called when the calculation of the given pk is terminated
+        Callback to be called when the process of the given pk is terminated
 
-        :param pk: the pk of the calculation
-        :param callback: the function to be called upon calculation termination
+        :param pk: the pk of the process
+        :param callback: the function to be called upon process termination
         """
-        calculation = load_node(pk=pk)
-        self._poll_calculation(calculation, callback)
+        process = load_node(pk=pk)
+        self._poll_process(process, callback)
 
-    def get_calculation_future(self, pk):
```

Comment:
I am not very sure but I think _poll_calculation is the only mechanism here rather than the backup for the rmq broadcast. Since the `on_process_finished` callback is only registered in `_poll_calculation` but not to rmq.
But for sure, `get_calculation_futuire` uses rmq broker and sets poll mechanism as backup, however this method is not called elsewhere in the code base. 

Or did I get it wrong?